apollo-ios/Sources/Apollo/GraphQLExecutor.swift

```diff
@@ -191,10 +197,51 @@ final class GraphQLExecutor<Source: GraphQLExecutionSource> {
     variables: GraphQLOperation.Variables? = nil,
     accumulator: Accumulator
   ) throws -> Accumulator.FinalResult {
-    let info = ObjectExecutionInfo(
-      rootType: SelectionSet.self,
+    return try execute(
+      selectionSet: selectionSet,
+      on: data,
+      withRootCacheReference: root,
       variables: variables,
       schema: SelectionSet.Schema.self,
+      accumulator: accumulator
+    )
+  }
+
+  func execute<
```

Comment:
It feels like we could really still just use the one execution function above if we just change it from needed `RootSelectionSet` to just `SelectionSet`. It doesn't look like there is any other actual deviation in functionality between these two functions. Am I missing something?

Reply:
OK. I've made a note that we should chat about it this week and if there are changes I'll make a separate PR for that.